### DISCUSSION TITLE
How does "Also switch the foreground program" work?

### DISCUSSION THREAD

**Discussion by YukonDanger** [Belphemur/SoundSwitch#669] [Q&A] [answered]
First, thanks for making this very handy application!

For example, I have a video playing in my browser (foreground program?) while chatting on discord. Both are coming out of my headset.
I want to switch discord from my headset to my speakers but leave the browser playing from my headset (do not switch the browser)

Is that what unchecking the "Also switch the foreground program" option should accomplish? 

Cheers,

**Comment by Belphemur** [comment, marked as answer]
Hey,

Foreground program mean the application currently selected, that you are currently interacting with.

So the concept behind "Also switch foreground program" is to not only change the default audio device but also force the application currently in use to have its audio device changed.

I've disabled by default since this option can lead to some issues with application not properly switching audio device.

Currently, there is no way to do exactly what you're asking with SoundSwitch.

However, you could do this: 
 * Go in discord settings and change what audio device it uses instead of relying on the default. (Set the headset)
 * Use SoundSwitch for switching the default audio device that is used by the browser.
